test/wdio/conditional-rerender/cmp.test.tsx

```diff
@@ -0,0 +1,41 @@
+import { Fragment, h } from '@stencil/core';
+import { render } from '@wdio/browser-runner/stencil';
+
+const css = `main {
+  background: rgb(0, 0, 0);
+  padding: 30px;
+}
+header {
+  background: rgb(230, 230, 230);
+  padding: 30px;
+}
+section {
+  background: rgb(160, 160, 160);
+  padding: 30px;
+}
+footer {
+  background: rgb(100, 100, 100);
+  padding: 30px;
+}
+nav {
+  background: rgb(50, 50, 50);
+  padding: 30px;
+}`;
+
+describe('conditional-rerender', function () {
+  beforeEach(() => {
+    render({
+      template: () => (
+        <>
+          <style>{css}</style>
+          <conditional-rerender-root></conditional-rerender-root>
+        </>
+      ),
+    });
+  });
+
+  it('contains a button as a child', async () => {
+    await $('main').waitForExist();
```

Comment:
Previously, we had been using a [`setTimeout` call](https://github.com/ionic-team/stencil/pull/5456/files#diff-e96ae002fbd5cf28368a0849203fec35d2d7d7a4d19f86b3724090fe39cbb427L15) with a value of 500 to ensure that the `setTimeout` call in the component (which uses a value of 20). IIUC, `waitForExist` has a default wait time of 500ms. I know that 20ms is a _short_ amount of time, but is it possible that this call ends before that `setTimeout` call in the component completes?

Reply:
Actually I will remove this as the async `expect` call will do some retries until it hits the right content. I don't think we will run into a race condition here. The timeout is just too small.